lib/compiler/src/compile.erl

```diff
@@ -673,6 +674,7 @@ standard_passes() ->
      {iff,debug_info,?pass(save_abstract_code)},
 
      ?pass(expand_records),
+     ?pass(collect_records),
```

Comment:
I wonder if it would be preferred to change [expand_records](https://github.com/richcarl/otp/blob/58d4684881047547230e75857020a59023ac6a3c/lib/compiler/src/compile.erl#L1262-L1264) to return both the new code and the collected records. Otherwise this is the third time the AST will be traversed only for the purpose of collecting records (erl_lint, expand_records and collect_records).